pkg/controller/utils/machine_helper_test.go

```diff
@@ -0,0 +1,280 @@
+
+		objs := []runtime.Object{machineList}
+		mocks := testutils.NewTestMock(t, objs)
+		// Validate 2 LB for machine
+
+		// each Machine ought to have 2 NLBs at the start, so let's check
+		for _, machine := range machineList.Items {
+			machineInfo := types.NamespacedName{
+				Name:      machine.GetName(),
+				Namespace: machine.GetNamespace(),
+			}
+			// reload the object to make sure we're not just working with the "in-memory"
+			// representation, that being, the un-saved one.
+			err := mocks.FakeKubeClient.Get(context.TODO(), machineInfo, &machine)
+			if err != nil {
+				t.Fatalf("Couldn't reload the test machine (named %s): %v", machineInfo.Name, err)
+			}
+			l, lbNames, lbTypes, err := testutils.ValidateMachineLB(&machine)
+			if err != nil {
+				t.Fatalf("Couldn't lookup the LB info: %v", err)
+			}
+			if l != 2 {
+				t.Fatalf("Before the test we expect to have 2 load balancers, but got %d", l)
+			}
+			// check for our LB by name and type unless we need it to be missing
+			if !test.shouldFail {
+				found := false
+				for i := 0; i < l; i++ {
+					if lbNames[i] == test.nameToRemove && lbTypes[i] == awsproviderapi.NetworkLoadBalancerType {
+						found = true
+					}
+				}
+				if !found {
+					t.Fatalf("Machine %s doesn't have a network load balancer named %s. It has %s", machine.GetName(), test.nameToRemove, lbNames)
+				}
+			}
+		}
+
+		// Make change
+		err := RemoveAWSLBFromMasterMachines(mocks.FakeKubeClient, test.nameToRemove, machineList)
+		if err != nil {
+			if !test.shouldFail {
+				t.Fatalf("Unexpected test couldn't remove LB %s from Machine: %v", test.nameToRemove, err)
+			}
+		}
+
+		// Validate test.nameToRemove is missing
+
+		for _, machine := range machineList.Items {
+			// reload the object to make sure we're not just working with the "in-memory"
+			// representation, that being, the un-saved one.
+			machineInfo := types.NamespacedName{
+				Name:      machine.GetName(),
+				Namespace: machine.GetNamespace(),
+			}
+
+			err = mocks.FakeKubeClient.Get(context.TODO(), machineInfo, &machine)
+			if err != nil {
+				t.Fatalf("Couldn't reload the test machine (named %s): %v", machineInfo.Name, err)
+			}
+			l, lbNames, _, err := testutils.ValidateMachineLB(&machine)
+			if err != nil {
+				t.Fatalf("Couldn't load the LB info for %s: %v", machineInfo.Name, err)
+			}
+			var expectedCount int
+			if test.shouldFail {
+				expectedCount = 2
+			} else {
+				expectedCount = 1
+			}
+			if l != expectedCount {
+				t.Fatalf("Test to remove %s (should fail = %t) Expected to have %d LB, but got %d", test.nameToRemove, test.shouldFail, expectedCount, l)
+			}
+			found := false
+			for _, lbName := range lbNames {
+				if lbName == test.nameToRemove {
+					found = true
+				}
+			}
+			if found {
+				t.Fatalf("Machine %s still has load balancer named %s", machineInfo.Name, test.nameToRemove)
+			}
+		}
+	}
+}
```

Comment:
@2uasimojo has submitted a PR (awaiting the results of live testing) to change the RemoveAWSLBFromMasterMachines function to check for a match of the LB short name, instead of checking for the lblist to contain an LB with the same prefix as the short name we've found in the machine's provider spec (https://github.com/openshift/cloud-ingress-operator/pull/51)

originally I wrote the function to use strings.HasPrefix (line 30) because the short name would be something like "cluster-name-hash-ext" and the lb.Name would be "cluster-name-hash-ext-anotherhash-domain.example.com". however, if Eric's PR stands up to a live test (which is on my to-do list), we would probably want to write a test case where a machine has two LBs, one of which is named something that is the prefix of the other, in order to make sure that only the first gets removed, as expected.

for example, we'd name the LBs testlb-12345-ext and testlb-12345-ext-to-keep, and then attempt to remove just testlb-12345-ext. this sort of case is likely not to happen in the real world, which is why strings.HasPrefix has worked so far, but Eric's solution is better (assuming it passes live testing), as the current implementation would remove both LBs in this case.

Reply:
Maybe we can address that in a rebased #51?